packages/cli/src/deploy/common.ts

```diff
@@ -119,3 +119,10 @@ export type Module = DeterministicContract & {
    */
   readonly optional?: boolean;
 };
+
+export type CommonDeployOptions = {
+  readonly client: Client<Transport, Chain | undefined, Account>;
```

Comment:
in which cases do we have an undefined chain? couldn't we enforce it to be defined so we can remove chainId as an additional arg? (and just get it with client.chain.id) not sure if this makes sense